pkg/sql/catalog/lease/descriptor_state.go

```diff
@@ -162,24 +171,38 @@ func (t *descriptorState) upsertLeaseLocked(
 	// released! This is because the new lease is valid at the same desc
 	// version at a greater expiration.
 	s.mu.expiration = expiration
+	s.mu.session = session
 	toRelease = s.mu.lease
 	s.mu.lease = &storedLease{
 		prefix:     regionEnumPrefix,
 		id:         desc.GetID(),
 		version:    int(desc.GetVersion()),
 		expiration: storedLeaseExpiration(expiration),
+		sessionID:  []byte(session.ID()),
```

Comment:
The above if statement implies that `session` could be nil. Do we need to worry about nil dereferencing here?